rig/place_and_route/wrapper.py

```diff
@@ -1,15 +1,131 @@
+    place : function (Default: :py:func:`rig.place_and_route.place`)
+        **Optional.** Placement algorithm to use.
+    place_kwargs : dict (Default: {})
+        **Optional.** Algorithm-specific arguments for the placer.
+    allocate : function (Default: :py:func:`rig.place_and_route.allocate`)
+        **Optional.** Allocation algorithm to use.
+    allocate_kwargs : dict (Default: {})
+        **Optional.** Algorithm-specific arguments for the allocator.
+    route : function (Default: :py:func:`rig.place_and_route.route`)
+        **Optional.** Routing algorithm to use.
+    route_kwargs : dict (Default: {})
+        **Optional.** Algorithm-specific arguments for the router.
+    core_resource : resource (Default: :py:data:`~rig.machine.Cores`)
+        **Optional.** The resource identifier used for cores.
+    sdram_resource : resource (Default: :py:data:`~rig.machine.SDRAM`)
+        **Optional.** The resource identifier used for SDRAM.
+
+    Returns
+    -------
+    placements : {vertex: (x, y), ...}
+        A dictionary from vertices to the chip coordinate produced by
+        placement.
+    allocations : {vertex: {resource: slice, ...}, ...}
+        A dictionary from vertices to the resources allocated to it. Resource
+        allocations are dictionaries from resources to a :py:class:`slice`
+        defining the range of the given resource type allocated to the vertex.
+        These :py:class:`slice` objects have `start` <= `end` and `step` set to
+        None.
+    application_map : {application: {(x, y): set([core_num, ...]), ...}, ...}
+        A dictionary from application to the set of cores it should be loaded
+        onto. The set of cores is given as a dictionary from chip to sets of
+        core numbers.
+    routing_tables : {(x, y): \
+                      [:py:class:`~rig.routing_table.RoutingTableEntry`, \
+                       ...], ...}
+        The generated routing tables. Provided as a dictionary from chip to a
+        list of routing table entries.
+    """
+    # Place/Allocate/Route
+    placements = place(vertices_resources, nets, machine, constraints,
+                       **place_kwargs)
+    allocations = allocate(vertices_resources, nets, machine, constraints,
+                           placements, **allocate_kwargs)
+    routes = route(vertices_resources, nets, machine, constraints, placements,
+                   allocations, core_resource, **route_kwargs)
+
+    # Build data-structures ready to feed to the machine loading functions
+    application_map = build_application_map(vertices_applications, placements,
+                                            allocations, core_resource)
+
+    # Build data-structures ready to feed to the machine loading functions
+    routing_tables = build_routing_tables(routes, net_keys)
+
+    return placements, allocations, application_map, routing_tables
 
 
 def wrapper(vertices_resources, vertices_applications,
```

Comment:
Very irritatingly there is not (though of course there are 100s of slightly different and often broken ones floating around the web). Additionally, warnings do not show up without `-Wa` on your Python interpreter... Sigh... Ah well! It is probably good enough given the entire userbase of Rig has been following this saga in person...